include/cadmium/engine/pdevs_simulator.hpp

```diff
@@ -216,20 +163,124 @@ namespace cadmium {
                     }
                 }
 
-                auto log_state = [](const typename model_type::state_type& s) -> std::string {
-                    std::ostringstream oss;
-                    oss << "State for model ";
-                    oss << boost::typeindex::type_id<model_type>().pretty_name();
-                    oss << " is ";
-                    oss << s;
-                    return oss.str();
-                };
-
-                LOGGER::template log<cadmium::logger::logger_state, decltype(log_state), const typename model_type::state_type&>(log_state, _model.state);
+                LOGGER::template log<cadmium::logger::logger_state, decltype(formatter::log_state), const typename model_type::state_type&>(formatter::log_state, _model.state);
             }
     //TODO: use enable_if functions to give access to read state and messages in debug mode
         };
 
+
```

Comment:
Please add a comment that this is a wip solution, we are going to do some changes later.

Reply:
Done.